sqlglot/dialects/spark2.py

```diff
@@ -160,8 +160,26 @@ class Spark2(Hive):
         exp.Pad: lambda self, e: _annotate_by_similar_args(
             self, e, "this", "fill_pattern", target_type=exp.DataType.Type.TEXT
         ),
+        exp.If: lambda self, e: self._annotate_by_args(
+            e, "true", "false", promote=True, coerce_text=True
+        ),
+        exp.Coalesce: lambda self, e: self._annotate_by_args(
+            e, "this", "expressions", promote=True, coerce_text=True
+        ),
     }
 
+    TEXT_COERCES_TO = {}
```

Comment:
Leave a comment here explaining that this mapping is supports only the non-ANSI setting.